src/niscope/metadata/enums_addon.py

```diff
@@ -139,37 +140,1084 @@
+                'value': 'VAL_RTSI_0',
+            },
+            {
+                'name': 'RTSI_1',
+                'value': 'VAL_RTSI_1',
+            },
+            {
+                'name': 'RTSI_2',
+                'value': 'VAL_RTSI_2',
+            },
+            {
+                'name': 'RTSI_3',
+                'value': 'VAL_RTSI_3',
+            },
+            {
+                'name': 'RTSI_4',
+                'value': 'VAL_RTSI_4',
+            },
+            {
+                'name': 'RTSI_5',
+                'value': 'VAL_RTSI_5',
+            },
+            {
+                'name': 'RTSI_6',
+                'value': 'VAL_RTSI_6',
+            },
+            {
+                'name': 'PFI_0',
+                'value': 'VAL_PFI_0',
+            },
+            {
+                'name': 'PFI_1',
+                'value': 'VAL_PFI_1',
+            },
+            {
+                'name': 'PFI_2',
+                'value': 'VAL_PFI_2',
+            },
+            {
+                'name': 'PXI_STAR',
+                'value': 'VAL_PXI_STAR',
+            },
+            {
+                'name': 'AUX_0_PFI_0',
+                'value': 'VAL_AUX_0_PFI_0',
+            },
+            {
+                'name': 'AUX_0_PFI_1',
+                'value': 'VAL_AUX_0_PFI_1',
+            },
+            {
+                'name': 'AUX_0_PFI_2',
+                'value': 'VAL_AUX_0_PFI_2',
+            },
+            {
+                'name': 'AUX_0_PFI_3',
+                'value': 'VAL_AUX_0_PFI_3',
+            },
+            {
+                'name': 'AUX_0_PFI_4',
+                'value': 'VAL_AUX_0_PFI_4',
+            },
+            {
+                'name': 'AUX_0_PFI_5',
+                'value': 'VAL_AUX_0_PFI_5',
+            },
+            {
+                'name': 'VAL_AUX_0_PFI_6',
+                'value': 'VAL_AUX_0_PFI_6',
+            },
+            {
+                'name': 'VAL_AUX_0_PFI_7',
+                'value': 'VAL_AUX_0_PFI_7',
+            },
+        ],
+    },
+    'TriggerSource': {
+        'values': [
+            {
+                'name': 'Channel_0',
+                'value': 0,
+            },
+            {
+                'name': 'Channel_1',
+                'value': 1,
+            },
+            {
+                'name': 'Channel_2',
+                'value': 2,
+            },
+            {
+                'name': 'Channel_3',
+                'value': 3,
+            },
+            {
+                'name': 'Channel_4',
+                'value': 4,
+            },
+            {
+                'name': 'Channel_5',
+                'value': 5,
+            },
+            {
+                'name': 'Channel_6',
+                'value': 6,
+            },
+            {
+                'name': 'Channel_7',
+                'value': 7,
+            },
+            {
+                'name': 'VAL_EXTERNAL',
+                'value': 'VAL_EXTERNAL',
```

Comment:
Why is this one a string when the others are numbers?

Reply:
I could not see TriggerSource defined in niscope.h. I took these values form LabVIEW, According to LabVIEW control this rest all are numbers while VAL_EXTERNAL is string. 

VAL_EXTERNAL is defined as string in iviscope header file aswell.

Reply:
From an API perspective, the function expects a single type. Where this particular enum is used `ConfigureTriggerWindow` it needs to be a string.

Another option is to look at the actual LV VI. While the values are the numbers you have listed, they are passed in as strings, not integers.

Looking in niScope.h, I do see a list of VAL's for trigger source, but the channels aren't in that list of defined values. So for this, there are a couple of things:
1. Make all the values strings: '1', '2', etc.
1. Make the names look like they came from the C API, even though they didn't really: 'NISCOPE_VAL_CHANNEL_1', etc

There are a lot of entries in the TRIGGER_SOURCE list in niScope.h, so I see two options here. I don't have a strong preference which way you go.
1. Rename this enum to be specific to trigger type window. `WindowTriggerSource`?
1. Leave the name generic so it can be used else where, but add all the other sources from the header file.

I see pros and cons with either solution.

Reply:
I have fixed this by adding all enums from .h file, I have also fixed the names

Reply:
What we had done in NI-FGEN and NI-DCPower is to **not** provide string-type enums for terminals, since these APIs participate in system-wide routing, and providing enums limit routing options.

We want customers to route between devices by using strings such as "/TheOtherDevice/MarkerEvent2"